src/ray/raylet/worker.cc

```diff
@@ -110,10 +107,15 @@ void Worker::AcquireTaskCpuResources(const ResourceIdSet &cpu_resources) {
   task_resource_ids_.Release(cpu_resources);
 }
 
+void Worker::SetGetTaskReplyAndCallback(rpc::GetTaskReply *reply,
+                                        rpc::SendReplyCallback send_reply_callback) {
+  reply_ = reply;
```

Comment:
Can we add a check here that both of these are `nullptr`? Also, is it true that this should only be called if `UsePush()`?

Reply:
Yes, it's sure that they are both nullptr.
And this function is only called when UsePush() returns false. We should not call send_reply_callback here because we want to block the raylet client until the task has been assigned to the worker. So we save send_reply_callback here and call it in AssignTask function.